### PR TITLE
Update thread/threading/concurrent from CPython 3.10.5

### PR DESCRIPTION
**Pull request by youknowone**
```
======================================================================
ERROR: test_interrupt_main_invalid_signal (test.test_threading.InterruptMainTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/user/Projects/RustPython/vm/pylib-crate/Lib/test/test_threading.py", line 1692, in test_interrupt_main_invalid_signal
    self.assertRaises(ValueError, _thread.interrupt_main, -1)
AttributeError: module '_thread' has no attribute 'interrupt_main'
```

### CODE REVIEW

**Comment by fanninpm** [comment]
Any update on this, now that #4082 has been merged?

File: Lib/test/test_threading.py [hunk 1]
```diff
@@ -1375,13 +1632,40 @@ class MiscTestCase(unittest.TestCase):
...
+
+            signal.signal(signum, signal.SIG_DFL)
+            _thread.interrupt_main(signum)
+        finally:
+            # Restore original handler
+            signal.signal(signum, handler)
+
+    # TODO: RUSTPYTHON
+    @unittest.expectedFailure
     def test_interrupt_main_subthread(self):
```
Comment (fanninpm) [inline Lib/test/test_threading.py:1669]: Did this raise an error on your machine?

Comment (youknowone) [reply Lib/test/test_threading.py]: Yes, it does.

Comment (youknowone) [reply Lib/test/test_threading.py]: Oh, this is flaky.

**Pull request merged**